app/controllers/api/assignments_controller.rb

```diff
@@ -5,50 +5,56 @@ class Api::AssignmentsController < ApplicationController
-                        created_assignments.append(assignment)
-                    else 
-                        action_item.destroy
-                        created_action_items.each {|item| item.destroy}
-                        Raven.capture_message("Could not create action item")
-                        render json: { error: 'Could not create action item' }, status: :unprocessable_entity
-                        return
-                    end
+        begin
+            due_date = DateTime.parse(single_assignment_params.fetch(:due_date))
+        rescue ArgumentError
+            due_date = nil
+        end 
+        action_item_params = {
+            title:  single_assignment_params.fetch(:title),
+            description: single_assignment_params.fetch(:description),
+            due_date: due_date,
+            category: single_assignment_params.fetch(:category),
+            file: single_assignment_params.fetch(:file),
+        }
+        if single_assignment_params.fetch(:file).eql?("null") || single_assignment_params.fetch(:file).eql?("undefined") || !single_assignment_params.fetch(:file).present?
+            action_item = ActionItem.new(action_item_params.except(:due_date, :file)) 
+        else 
+            action_item = ActionItem.new(action_item_params.except(:due_date)) 
+        end
+        template_sentry_helper(action_item)
+        action_item[:is_template] = false
+        if !participant_ids.empty? && action_item.save
+            prepare_bulk_assignment(participant_ids, action_item, due_date).each do |assignment|
+                assignment_sentry_helper(assignment)  
+                if assignment.save
+                    #AssignmentMailer.with(assignment: assignment, action_item: action_item).new_assignment.deliver_now
+                    puts "sent"
+                    puts assignment
+                else 
+                    action_item.destroy
+                    created_action_items.each {|item| item.destroy}
+                    Raven.capture_message("Could not create action item")
+                    render json: { error: 'Could not create action item' }, status: :unprocessable_entity
+                    return
                 end
-            else
-                created_action_items.each {|item| item.destroy}
-                Raven.capture_message("Could not create action item")
-                render json: { error: 'Could not create action item' }, status: :unprocessable_entity
-                return
             end
+        else
+            created_action_items.each {|item| item.destroy}
+            Raven.capture_message("Could not create action item")
+            render json: { error: 'Could not create action item' }, status: :unprocessable_entity
+            return
         end
-        render json: created_assignments, status: :created
-    end 
+    #what is render json supposed to do?
```

Comment:
We'd need this uncommented. `render` is the function called to give a response in this case.  json is the format the response is getting sent back as and `created_assignments` was the variable getting serialized as json and sent as a response. We need a response sent back on successful creation  in correct format for Calvin's PR to work